### PR TITLE
Wizard data

### PR DESCRIPTION
**Pull request by Jacksonzhang34**
**Issue #, if available:**

**Description of changes:**

**Why is this change necessary:**
Wizard Data used for getting and setting config key values 
**How was this change tested:**
tested through unit testing of all getter and setter methods 
**Any additional information or context required to review the change:**

**Checklist:**
- [ ] Updated the README if applicable
- [ ] Updated or added new unit tests
- [ ] Updated or added new integration tests
- [ ] Updated or added new end-to-end tests
- [ ] If your code makes a remote network call, it was tested with a proxy
- [ ] You confirm that the change is backwards compatible

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: gdk/wizard/WizardData.py [hunk 1]
```diff
@@ -0,0 +1,124 @@
+from gdk.wizard.ConfigEnum import ConfigEnum
+
+
+class WizardData:
+    def __init__(self, field_dict):
+        self.field_dict = field_dict
+        self.project_config = self.field_dict[ConfigEnum.COMPONENT.value.key]
+        self.component_name = next(iter(self.project_config))
```
Comment (saranyailla) [inline gdk/wizard/WizardData.py:8]: shouldn't these be reversed?

File: gdk/wizard/WizardData.py [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+
+    def get_author(self):
+        try:
+            return self.project_config.get(
+                self.component_name, ConfigEnum.COMPONENT_NAME.value.default
+            )[ConfigEnum.AUTHOR.value.key]
```
Comment (saranyailla) [inline gdk/wizard/WizardData.py:14]: Use `dict.get(key,default)` instead of using default value after `KeyError` is raised.

Comment (Jacksonzhang34) [reply gdk/wizard/WizardData.py:14]: gotcha so remove all the try, except and replace it with dict.get right

Comment (saranyailla) [reply gdk/wizard/WizardData.py:14]: `self.project_config.get(self.component_name)` is going to be a dictionary object. Store it once and use it internally within the class.

Comment (Jacksonzhang34) [reply gdk/wizard/WizardData.py:14]: got it fixing right now

File: gdk/wizard/ConfigEnum.py [hunk 3]
```diff
@@ -16,12 +16,12 @@ class ConfigEnum(Enum):
     AUTHOR = ConfigEnumDefault("author", default="gg-customer")
     VERSION = ConfigEnumDefault("version", default="1.0.0")
 
-    BUILD = ConfigEnumDefault("build")
+    BUILD = ConfigEnumDefault("build", default="build")
```
Comment (saranyailla) [inline gdk/wizard/ConfigEnum.py:19]: Isn't the default value - `{}`

File: gdk/wizard/WizardData.py [hunk 4]
```diff
@@ -0,0 +1,120 @@
+from gdk.wizard.ConfigEnum import ConfigEnum
+
+
+class WizardData:
+    def __init__(self, field_dict):
+        self.field_dict = field_dict
+        self.project_config = self.field_dict[ConfigEnum.COMPONENT.value.key]
+        self.component_name = next(iter(self.project_config))
+        self.component_name_fields = self.project_config.get(
```
Comment (saranyailla) [inline gdk/wizard/WizardData.py:9]: Not a blocker. Just a naming suggestion - `self.component_config`.

File: gdk/wizard/WizardData.py [hunk 5]
```diff
@@ -0,0 +1,112 @@
...
+
+    def set_build_options(self, value):
+        self._set_build_config_values(ConfigEnum.BUILD_OPTIONS.value, value)
+
+    def _set_publish_config_values(self, field, value):
+        if value:
+            self.component_config[ConfigEnum.PUBLISH.value.key][field.key] = value
+
+    def set_bucket(self, value):
+        self._set_publish_config_values(ConfigEnum.BUCKET.value, value)
```
Comment (saranyailla) [inline gdk/wizard/WizardData.py:102]: `ConfigEnum.BUCKET.value.key`

File: gdk/wizard/WizardData.py [hunk 6]
```diff
@@ -0,0 +1,112 @@
...
+
+    def set_region(self, value):
+        self._set_publish_config_values(ConfigEnum.REGION.value, value)
```
Comment (saranyailla) [inline gdk/wizard/WizardData.py:105]: `ConfigEnum.REGION.value.key`

File: gdk/wizard/WizardData.py [hunk 7]
```diff
@@ -0,0 +1,112 @@
...
+
+    def set_publish_options(self, value):
+        self._set_publish_config_values(ConfigEnum.PUBLISH_OPTIONS.value, value)
```
Comment (saranyailla) [inline gdk/wizard/WizardData.py:108]: `ConfigEnum.PUBLISH_OPTIONS.value.key`

**Review by saranyailla — APPROVED** [review]

**Review by urvashijain18 — APPROVED** [review]

**Pull request merged**